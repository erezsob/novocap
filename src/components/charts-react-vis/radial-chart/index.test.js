import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { RadialChart } from 'react-vis';
import CustomRadialChart from './index';
import ChartTitle from '../../chart-title';

const data = {
  drinks: [
    {
      x: '1',
      y: 100
    }
  ],
  food: [
    {
      x: '1',
      y: 1200
    }
  ],
  other: [
    {
      x: '1',
      y: 300
    }
  ]
};

describe('<CustomRadialChart />', () => {
  const wrapper = shallow(<CustomRadialChart data={data} />);

  test('renders properly', () => {
    const tree = renderer.create(<CustomRadialChart data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders a <ChartTitle />', () => {
    expect(wrapper.find(ChartTitle)).toHaveLength(1);
  });

  test('renders a <RadialChart /> with the right properties', () => {
    expect(wrapper.find(RadialChart)).toHaveLength(1);
    expect(wrapper.find(RadialChart).props().showLabels).toBeTruthy();
    expect(wrapper.find(RadialChart).props().animation).toBeTruthy();
  });
});
