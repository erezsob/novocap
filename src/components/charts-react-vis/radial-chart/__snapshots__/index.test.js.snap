// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomRadialChart /> renders properly 1`] = `
<div>
  <div
    className="label"
  >
    Sales
  </div>
  <div
    className="container"
  >
    <div
      className="rv-xy-plot chart rv-radial-chart"
      style={
        Object {
          "height": "250px",
          "width": "250px",
        }
      }
    >
      <svg
        className="rv-xy-plot__inner"
        height={250}
        onClick={[Function]}
        onDoubleClick={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onWheel={[Function]}
        style={undefined}
        width={250}
      >
        <g
          className="rv-xy-plot__series--arc__animation-wrapper"
        >
          <g
            className="rv-xy-plot__series rv-xy-plot__series--arc chart"
            onClick={[Function]}
            onContextMenu={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            opacity={1}
            pointerEvents="none"
            transform="translate(125,125)"
          >
            <path
              className="rv-xy-plot__series rv-xy-plot__series--arc-path rv-radial-chart__series--pie__slice "
              d="M7.34788079488412e-15,-120L6.735557395310443e-15,-110Z"
              onClick={[Function]}
              onContextMenu={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "fill": "#FFD968",
                  "opacity": 1,
                  "stroke": "#FFD968",
                }
              }
            />
            <path
              className="rv-xy-plot__series rv-xy-plot__series--arc-path rv-radial-chart__series--pie__slice "
              d="M7.34788079488412e-15,-120L6.735557395310443e-15,-110Z"
              onClick={[Function]}
              onContextMenu={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "fill": "#00A3E0",
                  "opacity": 1,
                  "stroke": "#00A3E0",
                }
              }
            />
            <path
              className="rv-xy-plot__series rv-xy-plot__series--arc-path rv-radial-chart__series--pie__slice "
              d="M7.34788079488412e-15,-120L6.735557395310443e-15,-110Z"
              onClick={[Function]}
              onContextMenu={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "fill": "#E51743",
                  "opacity": 1,
                  "stroke": "#E51743",
                }
              }
            />
          </g>
          <g
            className="rv-xy-plot__series rv-xy-plot__series--arc chart"
            onClick={[Function]}
            onContextMenu={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            opacity={0}
            pointerEvents="all"
            transform="translate(125,125)"
          >
            <path
              className="rv-xy-plot__series rv-xy-plot__series--arc-path rv-radial-chart__series--pie__slice "
              d="M7.34788079488412e-15,-120L6.735557395310443e-15,-110Z"
              onClick={[Function]}
              onContextMenu={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "fill": "#FFD968",
                  "opacity": 1,
                  "stroke": "red",
                }
              }
            />
            <path
              className="rv-xy-plot__series rv-xy-plot__series--arc-path rv-radial-chart__series--pie__slice "
              d="M7.34788079488412e-15,-120L6.735557395310443e-15,-110Z"
              onClick={[Function]}
              onContextMenu={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "fill": "#00A3E0",
                  "opacity": 1,
                  "stroke": "red",
                }
              }
            />
            <path
              className="rv-xy-plot__series rv-xy-plot__series--arc-path rv-radial-chart__series--pie__slice "
              d="M7.34788079488412e-15,-120L6.735557395310443e-15,-110Z"
              onClick={[Function]}
              onContextMenu={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "fill": "#E51743",
                  "opacity": 1,
                  "stroke": "red",
                }
              }
            />
          </g>
        </g>
        <g
          className="rv-xy-plot__series rv-xy-plot__series--label undefined"
          transform="translate(5,5)"
        >
          <text
            alignmentBaseline="text-before-edge"
            className="rv-xy-plot__series--label-text"
            fontSize="12px"
            onClick={[Function]}
            onContextMenu={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            textAnchor="end"
            transform="rotate(0,120,-12)"
            x={120}
            y={-12}
          >
            Other
          </text>
          <text
            alignmentBaseline="text-before-edge"
            className="rv-xy-plot__series--label-text"
            fontSize="12px"
            onClick={[Function]}
            onContextMenu={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            textAnchor="end"
            transform="rotate(0,120,-12)"
            x={120}
            y={-12}
          >
            Drinks
          </text>
          <text
            alignmentBaseline="text-before-edge"
            className="rv-xy-plot__series--label-text"
            fontSize="12px"
            onClick={[Function]}
            onContextMenu={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            textAnchor="end"
            transform="rotate(0,120,-12)"
            x={120}
            y={-12}
          >
            Food
          </text>
        </g>
      </svg>
    </div>
    <div
      className="total"
    >
      NaN
    </div>
  </div>
</div>
`;
